tests/test_infraction.py

```diff
@@ -0,0 +1,174 @@
+import os
+from enum import IntEnum
+
+import ape
+import pytest
+from eth_account import Account
+from eth_account.messages import encode_defunct
+from hexbytes import HexBytes
+from web3 import Web3
+
+TIMEOUT = 1000
+MAX_DKG_SIZE = 31
+FEE_RATE = 42
+ERC20_SUPPLY = 10**24
+DURATION = 48 * 60 * 60
+ONE_DAY = 24 * 60 * 60
+
+RitualState = IntEnum(
+    "RitualState",
+    [
+        "NON_INITIATED",
+        "DKG_AWAITING_TRANSCRIPTS",
+        "DKG_AWAITING_AGGREGATIONS",
+        "DKG_TIMEOUT",
+        "DKG_INVALID",
+        "ACTIVE",
+        "EXPIRED",
+    ],
+    start=0,
+)
+
+
+# This formula returns an approximated size
+# To have a representative size, create transcripts with `nucypher-core`
+def transcript_size(shares, threshold):
+    return int(424 + 240 * (shares / 2) + 50 * (threshold))
+
+
+def gen_public_key():
+    return (os.urandom(32), os.urandom(32), os.urandom(32))
+
+
+def access_control_error_message(address, role=None):
+    role = role or b"\x00" * 32
+    return f"account={address}, neededRole={role}"
+
+
+@pytest.fixture(scope="module")
+def nodes(accounts):
+    return sorted(accounts[:MAX_DKG_SIZE], key=lambda x: x.address.lower())
+
+
+@pytest.fixture(scope="module")
+def initiator(accounts):
+    initiator_index = MAX_DKG_SIZE + 1
+    assert len(accounts) >= initiator_index
+    return accounts[initiator_index]
+
+
+@pytest.fixture(scope="module")
+def deployer(accounts):
+    deployer_index = MAX_DKG_SIZE + 2
+    assert len(accounts) >= deployer_index
+    return accounts[deployer_index]
+
+
+@pytest.fixture(scope="module")
+def treasury(accounts):
+    treasury_index = MAX_DKG_SIZE + 3
+    assert len(accounts) >= treasury_index
+    return accounts[treasury_index]
+
+
+@pytest.fixture()
+def application(project, deployer, nodes):
+    contract = project.ChildApplicationForCoordinatorMock.deploy(sender=deployer)
+    for n in nodes:
+        contract.updateOperator(n, n, sender=deployer)
+        contract.updateAuthorization(n, 42, sender=deployer)
+    return contract
+
+
+@pytest.fixture()
+def erc20(project, initiator):
+    token = project.TestToken.deploy(ERC20_SUPPLY, sender=initiator)
+    return token
+
+
+@pytest.fixture()
+def coordinator(project, deployer, application, erc20, initiator, oz_dependency):
```

Comment:
I believe this fixture is repeated in `test_coordinator`. More generally, other fixtures are being duplicated as well, e.g., `erc20(...)`, etc. You could consider putting common fixtures into the conftest.py to prevent duplicating code.

Reply:
they're all copied over from `test_coordinator`. I would expect all fixtures to be in `conftest` so I assumed there was a reason that were being kept in the same file as the tests that used them. If that's not the case, then I'm very much on board with moving everything to `conftest` and getting rid of all the duplicates

Reply:
In some cases, a mock contract is used instead of the actual contract because of specific testing needs. For example, see https://github.com/nucypher/nucypher-contracts/blob/main/tests/test_child_application.py#L56 for the `Coordinator` contract.

But other than those nuanced cases, moving common/generic fixtures to `conftest.py` should be fine and overridden as necessary in the individual test modules themselves, like the above.

To keep things simple, this modification does not need to occur in this PR; it could be done in a follow-up PR.

Reply:
opened an issue to track this https://github.com/nucypher/nucypher-contracts/issues/284